addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -1267,6 +1267,28 @@ def test_19_deliver_update_so_line_qty(self):
 
         self.assertEqual(len(sale_order.picking_ids), 2)
 
+    def test_update_so_line_qty_with_package(self):
+        """
+        Creates a sale order, then validates the delivery
+        modifying the sale order lines qty to 0
+        move line should be deleted.
+        """
+        self.product_a.type = 'product'
+        self.env['stock.quant']._update_available_quantity(
+            self.product_a, self.company_data['default_warehouse'].lot_stock_id, 10,
+            package_id=self.env['stock.quant.package'].create({'name': 'PacMan'}))
+
+        # Create sale order
+        sale_order = self._get_new_sale_order(product=self.product_a)
+        sale_order.action_confirm()
+
+        # Update the SO line
+        with Form(sale_order.with_context(import_file=True)) as so_form:
+            with so_form.order_line.edit(0) as line:
+                line.product_uom_qty = 0
+
+        self.assertFalse(sale_order.picking_ids.move_line_ids)
```

Comment:
Please add an assertion for the `package_level_id`! I am aware that the `move_line_ids` are linked, but it's advisable to include this check for better clarity.